src/devices/Ads1115/Ads1115.cs

```diff
@@ -151,14 +151,7 @@ public double RawToVoltage(short val)
                     break;
             }
 
-            if ((byte)_inputMultiplexer <= 0x03)
-            {
-                resolution = 65535.0;
-            }
-            else
-            {
-                resolution = 32768.0;
-            }
+            resolution = 32768.0;
```

Comment:
@TobBrandt there is also a note under that paragraph in the spec:

```
Single-endedsignalmeasurements,whereVAINN= 0 V and VAINP= 0 V to +FS,only usethe positivecoderangefrom0000hto 7FFFh.However,becauseof deviceoffset,theADS111xcan still outputnegativecodesin caseVAINPis closeto 0 V.
```

Should you be doing full range for single-ended readings? I'm wondering if the `(byte)_inputMultiplexer <= 0x03` wasn't the check for that but I'm not sure

Reply:
The way I read this note is as follows: for single ended inputs, the input is expected to be positive relative to ground. Therefore the raw value will only use the positive part of the range (0 to MAX_SHORT), but you can occasionally get small negative values due to measurement tolerances.
That still means you need to divide by MAX_SHORT for single ended inputs.

Btw, the single ended inputs have values  > 0x03 for the input multiplexer. The check is for differential inputs.

I’ll be doing a few measurements when I’m back at the office and post them here.

Reply:
Your change affects the differential inputs, which now divide by MAX_SHORT instead of MAX_USHORT. I haven't used the differential mode, but I'm quite sure the single-ended mode works as expected. And yes, it often does give a small negative number if you connect an input to ground, which indicates the sign handling is correct, otherwise a small negative (integer) number would be interpreted as something big.

Reply:
The original code worked correctly for the single ended mode. Just the differential mode is wrong by a factor of two.

I've made a few measurements with A0 = 3.3V, A1 = 5V, and gain = +-6.144V. Here are the result:

|InputMultiplexer|Raw value|Calculated Voltage|Actual Voltage|
| ---------------- | ----------- | ------- | ------------- |
|AIN0 = 0x04|0x4461|3.2821875|3.3|
|AIN1 = 0x05|0x6826|4.999125|5|
|AIN0_AIN1 = 0x00|0xDC76|-0.852951|-1.7|

Reply:
Seems to be you're right, then. AINO_AIN1 should calculate A0-A1 (If I read the doc correctly), which in your case is -1.7.

Reply:
So if I understand correctly you've made measurements as above after the fix and the actual voltage looks correct? If so I'm cool with merging this as is.

Reply:
Yes, I did.